Sources/BSON/Double+BSONValue.swift

```diff
@@ -1,6 +1,52 @@
 import NIO
 
 extension Double: BSONValue {
+    /*
+     * Initializes a `Double` from ExtendedJSON.
+     *
+     * Parameters:
+     *   - `json`: a `JSON` representing the canonical or relaxed form of ExtendedJSON for a `Double`.
+     *   - `keyPath`: an array of `String`s containing the enclosing JSON keys of the current json being passed in.
+     *              This is used for error messages.
+     *
+     * Returns:
+     *   - `nil` if the provided value is not a `Double`.
+     *
+     * Throws:
+     *   - `DecodingError` if `json` is a partial match or is malformed.
+     */
+    internal init?(fromExtJSON json: JSON, keyPath: [String]) throws {
+        switch json {
+        case let .number(n):
+            // relaxed extended JSON
+            guard let double = Double(exactly: n) else {
+                return nil
+            }
+            self = double
```

Comment:
since `n` is a `Double` already, we can just assign directly without doing the `Double(exactly:)` initializer.